test/metabase/query_processor_test/date_time_zone_functions_test.clj

```diff
@@ -809,120 +837,204 @@
          (t/format :iso-offset-date-time a)       ; a_dt_tz_text
          (t/format :iso-offset-date-time b)]))]]) ; b_dt_tz_text
 
+(mt/defdataset diff-time-zones-athena-cases
+  ;; This dataset contains the same set of values as [[diff-time-zones-cases]], but without the time zones.
+  ;; It is needed to test `datetime-diff` with Athena, since Athena supports `timestamp with time zone`
+  ;; in query expressions but not in a table. [[diff-time-zones-athena-cases-query]] uses this dataset
+  ;; to recreate [[diff-time-zones-cases]] for Athena as a query.
+  [["times"
+    [{:field-name "dt",      :base-type :type/DateTime}
+     {:field-name "dt_text", :base-type :type/Text}]
+    (for [dt [#t "2022-10-02T00:00:00"
+              #t "2022-10-02T01:00:00"
+              #t "2022-10-03T00:00:00"
+              #t "2022-10-09T00:00:00"
+              #t "2022-11-02T00:00:00"
+              #t "2023-01-02T00:00:00"
+              #t "2023-10-02T00:00:00"]]
+      [dt (u.date/format dt)])]])
+
+(def diff-time-zones-athena-cases-query
+  ;; This query recreates [[diff-time-zones-cases]] for Athena from [[diff-time-zones-athena-cases]].
+  (str/join "\n"
```

Comment:
Just a style nitpick: a multiline string might be more readable and easier to change:

```clojure
(def diff-time-zones-athena-cases-query
  ;; This query recreates [[diff-time-zones-cases]] for Athena from [[diff-time-zones-athena-cases]].
  "with x as (
     select
     with_timezone(dt, 'UTC') as dt
     , concat(dt_text, 'Z') as dt_text -- e.g. 2022-10-02T00:00:00Z
     , 'UTC' as time_zone
   from diff_time_zones_athena_cases.times
   union
   select
     with_timezone(dt, 'Africa/Lagos') as dt
     , concat(dt_text, '+01:00') as dt_text -- e.g. 2022-10-02T00:00:00+01:00
     , 'Africa/Lagos' as time_zone
   from diff_time_zones_athena_cases.times
   )
   select
     a.dt as a_dt_tz
     , a.dt_text as a_dt_tz_text
     , b.dt as b_dt_tz
     , b.dt_text as b_dt_tz_text
   from x a
   join x b on a.dt < b.dt and a.time_zone <> b.time_zone")
```

Reply:
Yeah. It adds extra whitespace, which is why I tried to avoid it but I think it's worth it for this length of query.